### PR TITLE
Implement api-design v0.2.0

### PR DESCRIPTION
**Pull request by mcanoy**
#### What does this PR do?

This PR allows the application to run for the first time. 
- It implements virtually all the ApiControllers (excludes BuildToolsApi). 
- It creates services to connect to the backend database via Spring Data/JPA/Postgres. 
- This creates a very modest amount of unit and integration tests. The integration tests connect to an in-memory db (H2).
- Incorporates Spring Cloud Kubernetes (from fabric8) that will allows to control our configuration management. (See sample yml in src/test/resources/sampledata)
- Uses maven, junk, mockito, spring fox and spring boot (1.4)
- Uses code generated from https://github.com/rht-labs/api-design  
#### Which tests illustrate how this code works?

Testing is currently minimally implemented. Currently they should all run using
`mvn test`. Further work will need to be done to separate the test types and expand the code coverage.

Integration tests to the db are annotated in the base class as @DataJPATest
#### Is there a relevant Issue open for this?

This is the initial PR
#### Are there any other relevant resources that should be reviewed as well?

The repo https://github.com/rht-labs/api-design is used for codegen and includes a script to run the codegen
#### Who would you like to review this?

/cc @mdanter @sherl0cks @oybed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                <artifactId>spring-boot-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>repackage</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin><!-- Needed only for deploying to Labs OpenShift from local box. Consider deleting - kmcanoy -->
```
Comment (sherl0cks) [inline pom.xml]: Please move this fabric8 plugin to a non-default maven profile

Comment (mcanoy) [reply pom.xml]: done

File: src/main/java/com/rhc/automation/Swagger2SpringBoot.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.rhc.automation.service.OpenShiftService;
+import com.rhc.automation.service.UserService;
+
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@SpringBootApplication
+@EnableSwagger2
+@ComponentScan(basePackages = "com.rhc.automation")
+public class Swagger2SpringBoot implements CommandLineRunner {
+    private static final Logger log = LoggerFactory.getLogger(Swagger2SpringBoot.class);
```
Comment (mdanter) [inline src/main/java/com/rhc/automation/Swagger2SpringBoot.java]: Please capitalize constants.

File: src/main/resources/META-INF/orm.xml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    <entity class="com.rhc.automation.model.Project" access="FIELD">
+        <attributes>
+            <id name="id">
+                <generated-value strategy="AUTO"/>
+            </id>
+            <basic name="buildEnvironment"/>
+            <basic name="promotionEnvironment"/>
+            <basic name="name"/>
+            <basic name="displayName"/>
+            <one-to-many name="apps" target-entity="com.rhc.automation.model.Application" fetch="EAGER"/>
```
Comment (mdanter) [inline src/main/resources/META-INF/orm.xml:123]: Please also implement corresponding many-to-one references in the xml.

**Review by mdanter — COMMENTED** [review]
Awesome!!!

File: src/main/java/com/rhc/automation/Swagger2SpringBoot.java (hunk 2 above)
Comment (mcanoy) [reply src/main/java/com/rhc/automation/Swagger2SpringBoot.java]: I removed this logger since it was not being used. Also removed other unused code

**Pull request merged**